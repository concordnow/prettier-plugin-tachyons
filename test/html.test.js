const fs = require("fs");
const path = require("path");
const prettier = require("prettier");
const plugin = require("..");

const fixture = fs.readFileSync(
  path.join(__dirname, "./__fixtures__/html.fix"),
  "utf-8"
);

function format(content, opts = {}) {
  return prettier.format(content, {
    ...opts,
    parser: "html",
    plugins: [plugin],
  });
}

describe("Test HTML parser", () => {
  test("basic html page", async () => {
    const formatted = format(fixture);
    expect(formatted).toMatchSnapshot();
  });
});
