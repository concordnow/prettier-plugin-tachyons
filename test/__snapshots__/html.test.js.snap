// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test HTML parser basic html page 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta charset=\\"utf-8\\" />
    <title>HTML test</title>
  </head>
  <body>
    <article class=\\"\\">
      <h1 class=\\"\\">Title</h1>
      <p class=\\"\\">
        Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy
        eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam
        voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet
        clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit
        amet.
      </p>
      <p class=\\"\\">
        Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy
        eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam
        voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet
        clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit
        amet.
      </p>
    </article>
  </body>
</html>
"
`;
